geopandas/array.py

```diff
@@ -518,15 +533,30 @@ def union(self, other):
     #
 
     def distance(self, other):
+        if self.crs and self.crs.is_geographic:
+            warnings.warn(
+                geographic_crs_msg.format("distance"), UserWarning, stacklevel=5,
+            )
         return self._binary_method("distance", self, other)
 
     def buffer(self, distance, resolution=16, **kwargs):
+        if self.crs and self.crs.is_geographic:
```

Comment:
Could this get condensed a little bit?
```
if isinstance(distance, (int, float)) and distance != 0 and self.crs and self.crs.is_geographic:
```

Reply:
I tried. This does not raise if you pass distance an array.